src/types/account-manager.ts

```diff
@@ -1,35 +1,61 @@
 import algosdk from 'algosdk'
-import {
-  getAccountAssetInformation,
-  getAccountInformation,
-  mnemonicAccountFromEnvironment,
-  multisigAccount,
-  randomAccount,
-  rekeyedAccount,
-} from '../account/account'
-import { getDispenserAccount } from '../account/get-dispenser-account'
-import { mnemonicAccount } from '../account/mnemonic-account'
-import { getKmdWalletAccount } from '../localnet/get-kmd-wallet-account'
-import { getLocalNetDispenserAccount } from '../localnet/get-localnet-dispenser-account'
-import { getSenderAddress, getSenderTransactionSigner } from '../transaction/transaction'
-import { AccountInformation, SigningAccount, TransactionSignerAccount } from './account'
+import { AccountInformation, DISPENSER_ACCOUNT, MultisigAccount, SigningAccount, TransactionSignerAccount } from './account'
 import { AlgoAmount } from './amount'
 import { ClientManager } from './client-manager'
-import { SendTransactionFrom } from './transaction'
+import { KmdAccountManager } from './kmd-account-manager'
 import LogicSigAccount = algosdk.LogicSigAccount
+import Account = algosdk.Account
+import TransactionSigner = algosdk.TransactionSigner
+import AccountInformationModel = algosdk.modelsv2.Account
+
+const memoize = <T = unknown, R = unknown>(fn: (val: T) => R) => {
+  const cache = new Map()
+  const cached = function (this: unknown, val: T) {
+    return cache.has(val) ? cache.get(val) : cache.set(val, fn.call(this, val)) && cache.get(val)
+  }
+  cached.cache = cache
+  return cached as (val: T) => R
+}
+
+/**
+ * Returns a `TransactionSigner` for the given account that can sign a transaction.
+ * This function has memoization, so will return the same transaction signer for a given account.
+ * @param account An account that can sign a transaction
+ * @returns A transaction signer
+ */
+export const getAccountTransactionSigner = memoize(function (
+  account: TransactionSignerAccount | Account | SigningAccount | LogicSigAccount | MultisigAccount,
+): TransactionSigner {
+  return 'signer' in account
+    ? account.signer
+    : 'lsig' in account
+      ? algosdk.makeLogicSigAccountTransactionSigner(account)
+      : algosdk.makeBasicAccountTransactionSigner(account)
+})
```

Comment:
This is copied from `getSenderTransactionSigner` function, but `SendTransactionFrom` is removed in lieu of specifying the types of account classes supported explicitly. This allows us to eventually deprecate it and means this file doesn't take any dependencies on legacy methods anymore.